pandas/tests/frame/test_misc_api.py

```diff
@@ -332,6 +332,15 @@ def test_deepcopy(self):
         for idx, value in compat.iteritems(series):
             self.assertNotEqual(self.frame['A'][idx], value)
 
+    def test_deepcopy_empty(self):
```

Comment:
> yeah this is better as def __copy__(self):.
> 
> also let's change the same in pandas/indexes/base.py. hopefully nothing will break.

Do you want both of those in this PR?



> add tests instead in test_generic.py

Sure, I can move it.  I put it here because I figured it ought to be near the other `deepcopy` test and didn't see any other obvious place for it.